src/machine learning analysis.py

```diff
@@ -0,0 +1,170 @@
+# author: Siqi Tao
+# date: 2021-11-26
+
+"""Downloads data as a csv file from online to a local filepath via a URL.
+Usage: download_script.py --url=<url> --filepath=<filepath> 
+ 
+Options:
+--input=<input>       The path or filename pointing to the data
+--prefix=<prefix>     The prefix where to write the output figure(s)/table(s) to and what to call it 
+"""
+
+import os
+import pandas as pd
+from docopt import docopt
+import string
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.compose import ColumnTransformer, make_column_transformer
+from sklearn.dummy import DummyClassifier, DummyRegressor
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.feature_selection import RFE, RFECV
+from sklearn.impute import SimpleImputer
+from sklearn.linear_model import LogisticRegression, Ridge, RidgeCV
+from sklearn.metrics import make_scorer
+from sklearn.model_selection import (
+    GridSearchCV,
+    RandomizedSearchCV,
+    ShuffleSplit,
+    cross_val_score,
+    cross_validate,
+    train_test_split,
+)
+from sklearn.pipeline import Pipeline, make_pipeline
+from sklearn.preprocessing import (
+    OneHotEncoder,
+    OrdinalEncoder,
+    PolynomialFeatures,
+    StandardScaler,
+)
+from sklearn.svm import SVC, SVR
+from scipy.stats import loguniform
+from sklearn.metrics import confusion_matrix
+import pickle
+
+opt = docopt(__doc__)
+
+def main(): 
+    
+    # Get data
+    input = opt["--input"]
+    train_df = pd.read_csv(f"{input}train.csv", low_memory=False)
+    test_df = pd.read_csv(f"{input}test.csv", low_memory=False)
+    
+    X_train = train_df.drop(columns=["FATALITY"])
+    X_test = test_df.drop(columns=["FATALITY"])
+    
+    y_train = train_df["FATALITY"]
+    y_test = test_df["FATALITY"]
+    
+    # Convert columns into string data type
+    cols = X_train.columns.tolist()
+    for i in range(len(cols)):
+        X_train[cols[i]] = X_train[cols[i]].astype(str)
+        X_test[cols[i]] = X_test[cols[i]].astype(str)
+    
+    #Transform columns (all categorical)
+    categorical_feats = cols
+    preprocessor = make_column_transformer(
+        (OneHotEncoder(handle_unknown="ignore", sparse=False), categorical_feats),
```

Comment:
extra comma at the end of the line (probably doesn't matter)

Reply:
Yeah I tried both. It doesn't matter